### PR TITLE
Static rewards and Integration Test

### PR DESCRIPTION
**Pull request by Joeysantoro**
Add a static rewards distributor and integration test the system against Fuse mainnet fork

### CODE REVIEW

File: src/rewards/FlywheelStaticRewards.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     @param rewards the rewards info for the market
+     */    
+    function setRewardsInfo(ERC20 market, RewardsInfo calldata rewards) external requiresAuth {
+        rewardsInfo[market] = rewards;
+        emit RewardsInfoUpdate(market, rewards.rewardsPerSecond, rewards.rewardsEndTimestamp);
+    }
+
+    /**
+     @notice calculate and transfer accrued rewards to flywheel core
+     @param market the market to accrue rewards for
```
Comment (thomas-waite) [inline src/rewards/FlywheelStaticRewards.sol:57]: Minor typo, `market the market`

File: src/test/FlywheelStaticRewardsTest.t.sol [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        (rewardsPerSecond, rewardsEndTimestamp) = rewards.rewardsInfo(market);
+
+        require(rewardsPerSecond == 1 ether);
+        require(rewardsEndTimestamp == newEnd);
+    }
+
+    function testGetAccruedRewards() public {
+        hevm.warp(1000);
+        testSetRewardsInfo();
```
Comment (thomas-waite) [inline src/test/FlywheelStaticRewardsTest.t.sol:41]: Didn't realise you could call the test functions like this, but makes sense! Sweet!

File: src/test/Integration.t.sol [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        // check market index
+        (uint224 index,) = flywheel.marketState(fTRIBE);
+        require(index == flywheel.ONE() + rewardsPerToken);
+
+        // claim and check user balance
+        flywheel.claim(user);
+        require(rewardToken.balanceOf(user) == userRewards);
+
+        // mint more tokens by user and rerun test
```
Comment (thomas-waite) [inline src/test/Integration.t.sol:96]: For readability, I'd perhaps split this second test - line 91 onwards - out into a seperate integration test function

**Review by thomas-waite — APPROVED** [review]
Looks great! Minor comments, love the Solidity integration test!

File: src/rewards/FlywheelStaticRewards.sol (hunk 1 above)
Comment (Joeysantoro) [reply src/rewards/FlywheelStaticRewards.sol:57]: the parameter is called `market` and the description follows

File: src/test/Integration.t.sol (hunk 3 above)
Comment (Joeysantoro) [reply src/test/Integration.t.sol:96]: this test kinda relies on both, I'm inclined to keep them together

**Pull request merged**